src/main/webapp/app/entities/programming-exercise/programming-exercise-dialog.controller.js

```diff
@@ -27,6 +27,7 @@
 
         function save () {
             vm.isSaving = true;
+            vm.programmingExercise.type = 'de.tum.in.www1.exerciseapp.domain.ProgrammingExercise';
```

Comment:
If you really can't avoid this approach, please make sure it is used in all other places (quiz exercises, model exercises) as well and properly document it)

Reply:
Done, I have also moved the type adding to the respective services, I guess this is more universal.